### PR TITLE
feat: refactor kotlin templates

### PR DESCRIPTION
**Pull request by everly-gif**
refactor kotlin templates

### CODE REVIEW

File: templates/kotlin/base/macros/methodNeedsSecurityParameters.twig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{% macro methodNeedsSecurityParameters(method) %}
```
Comment (lohanidamodar) [inline templates/kotlin/base/macros/methodNeedsSecurityParameters.twig:1]: if macros are used in only one place place it in that file only. Let's keep all macros that are used more than once together as utils.twig

**Review by lohanidamodar — CHANGES_REQUESTED** [review]

File: templates/kotlin/base/requests/location.twig [hunk 2]
```diff
@@ -0,0 +1,6 @@
+        return client.call(
```
Comment (abnegate) [inline templates/kotlin/base/requests/location.twig:1]: This template needs to import the `resultType` macro

File: templates/kotlin/base/macros/methodNeedsSecurityParameters.twig (hunk 1 above)
Comment (everly-gif) [reply templates/kotlin/base/macros/methodNeedsSecurityParameters.twig:1]: okay sure .

File: templates/kotlin/base/requests/location.twig (hunk 2 above)
Comment (everly-gif) [reply templates/kotlin/base/requests/location.twig:1]: thank you !

**Review by abnegate — APPROVED** [review]

**Review by lohanidamodar — APPROVED** [review]

**Pull request merged**